Userland/Libraries/LibC/stdlib.cpp

```diff
@@ -1049,11 +1049,15 @@ char* realpath(const char* pathname, char* buffer)
         return nullptr;
     }
     size_t size = PATH_MAX;
-    if (buffer == nullptr)
+    bool self_allocated = false;
+    if (buffer == nullptr) {
         buffer = (char*)malloc(size);
+        self_allocated = true;
+    }
     Syscall::SC_realpath_params params { { pathname, strlen(pathname) }, { buffer, size } };
     int rc = syscall(SC_realpath, &params);
     if (rc < 0) {
+        free(buffer);
```

Comment:
You forgot did not use `self_allocated`.

Reply:
Oh, nevermind. you added it in ed3410f but only used it from 7f666635. I didn't notice.

Reply:
Adding `bool self_allocated = false;` and `self_allocated = true;` in 7f66663 would make more sense in my opinion though.

Reply:
It's wrong nevertheless, since the new line 1060 doesn't check `self-allocated`. D'oh! :D